core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -461,15 +463,25 @@ private[log] class Cleaner(val id: Int,
         ByteBufferMessageSet.writeMessage(buffer, messageOffset.message, messageOffset.offset)
       MessageSet.messageSetSize(messagesIterable)
     } else {
+      val messageSetTimestamp = MessageSet.validateMagicValuesAndGetTimestamp(messages.map(_.message))
+      val firstAbsoluteOffset = messages.head.offset
       var offset = -1L
+      val magicValue = messages.head.message.magic
```

Comment:
I was thinking only applying the new message format to the messages appended after the change, so the log has a clear cut-off offset where all the new format comes after that. It is probably not necessary. What do you think?